// connectwise-rest@1.3.0/dist/Automate/AVTemplatePoliciesAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AVTemplatePoliciesAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Automate_1 = __importDefault(require("../Automate"));
/**
 * @module AVTemplatePoliciesAPI
 */
/**
 * AVTemplatePolicies module
 * @public
 */
class AVTemplatePoliciesAPI extends Automate_1.default {
    constructor(props) {
        super(props);
    }
    getAntivirusTemplatePolicyList(params = {}) {
        return this.request({
            path: `/api/v1/AVTemplatePolicies`,
            method: 'get',
            params,
        });
    }
    postAntivirusTemplatePolicy(AVTemplatePolicy) {
        return this.request({
            path: `/api/v1/AVTemplatePolicies`,
            method: 'post',
            data: AVTemplatePolicy,
        });
    }
    getAntivirusTemplatePolicyDataList(params = {}) {
        return this.request({
            path: `/api/v1/AVTemplatePolicyData`,
            method: 'get',
            params,
        });
    }
    postAntivirusTemplatePolicyData(AVTemplatePolicyData) {
        return this.request({
            path: `/api/v1/AVTemplatePolicyData`,
            method: 'post',
            data: AVTemplatePolicyData,
        });
    }
    getGroupList(params = {}) {
        return this.request({
            path: `/api/v1/Groups`,
            method: 'get',
            params,
        });
    }
    postGroup(Group) {
        return this.request({
            path: `/api/v1/Groups`,
            method: 'post',
            data: Group,
        });
    }
    getTemplateAvailablePropertyList(params = {}) {
        return this.request({
            path: `/api/v1/TemplateAvailableProperties`,
            method: 'get',
            params,
        });
    }
    postTemplateAvailableProperty(TemplateAvailableProperty) {
        return this.request({
            path: `/api/v1/TemplateAvailableProperties`,
            method: 'post',
            data: TemplateAvailableProperty,
        });
    }
    getTemplatePropertyList(params = {}) {
        return this.request({
            path: `/api/v1/TemplateProperties`,
            method: 'get',
            params,
        });
    }
    postTemplateProperty(TemplateProperty) {
        return this.request({
            path: `/api/v1/TemplateProperties`,
            method: 'post',
            data: TemplateProperty,
        });
    }
}
exports.AVTemplatePoliciesAPI = AVTemplatePoliciesAPI;
