// connectwise-rest@1.3.0/dist/Automate/ComputersAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComputersAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Automate_1 = __importDefault(require("../Automate"));
/**
 * @module ComputersAPI
 */
/**
 * Computers module
 * @public
 */
class ComputersAPI extends Automate_1.default {
    constructor(props) {
        super(props);
    }
    getComputerMenuList(params = {}) {
        return this.request({
            path: `/api/v1/ComputerMenus`,
            method: 'get',
            params,
        });
    }
    postComputerMenu(ComputerMenu) {
        return this.request({
            path: `/api/v1/ComputerMenus`,
            method: 'post',
            data: ComputerMenu,
        });
    }
    getComputerList(params = {}) {
        return this.request({
            path: `/api/v1/Computers`,
            method: 'get',
            params,
        });
    }
    getComputerAlerts(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Alerts`,
            method: 'get',
            params,
        });
    }
    getMaintenanceAlertSuspensionList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/AlertSuspensions/MaintenanceWindow`,
            method: 'get',
            params,
        });
    }
    getTemplateAlertSuspensionList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/AlertSuspensions/TemplateDiversion`,
            method: 'get',
            params,
        });
    }
    getComputerBios(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/bios`,
            method: 'get',
            params,
        });
    }
    getCommandExecuteList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/CommandExecute`,
            method: 'get',
            params,
        });
    }
    postCommandExecute(computerId, CommandExecute) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/CommandExecute`,
            method: 'post',
            data: CommandExecute,
        });
    }
    getCommandHistoryList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/CommandHistory`,
            method: 'get',
            params,
        });
    }
    getComputerPatchingPolicyListAsync(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/ComputerPatchingPolicies`,
            method: 'get',
            params,
        });
    }
    getComputerDeviceList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Devices`,
            method: 'get',
            params,
        });
    }
    getComputerDrivers(computerId, params = {}) {
        return this.request({
            path: `/api/v1/computers/${computerId}/drivers`,
            method: 'get',
            params,
        });
    }
    getComputerDriveSmartDataList(computerId, driveId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Drives/${driveId}/SmartData`,
            method: 'get',
            params,
        });
    }
    getComputerEffectivePatchingPolicy(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/EffectivePatchingPolicy`,
            method: 'get',
            params,
        });
    }
    getComputerMicrosoftUpdateDataListAsync(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/MicrosoftUpdates`,
            method: 'get',
            params,
        });
    }
    getMonitorAlertSuspensionList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/MonitorAlertSuspensions`,
            method: 'get',
            params,
        });
    }
    postSuspension(computerId, integer) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/MonitorAlertSuspensions`,
            method: 'post',
            data: integer,
        });
    }
    getComputerMonitorsList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Monitors`,
            method: 'get',
            params,
        });
    }
    getComputerOperatingSystem(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/OperatingSystem`,
            method: 'get',
            params,
        });
    }
    getDevicePatchingStatisticsAsync(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/PatchingStats`,
            method: 'get',
            params,
        });
    }
    getComputerPatchJobListAsync(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/PatchJobs`,
            method: 'get',
            params,
        });
    }
    getComputerPrintersList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Printers`,
            method: 'get',
            params,
        });
    }
    getComputerProcessorList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Processors`,
            method: 'get',
            params,
        });
    }
    getComputerRunningScripts(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/RunningScripts`,
            method: 'get',
            params,
        });
    }
    getComputerScheduledScripts(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/ScheduledScripts`,
            method: 'get',
            params,
        });
    }
    postComputerScheduledScript(computerId, ScheduledScript) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/ScheduledScripts`,
            method: 'post',
            data: ScheduledScript,
        });
    }
    getComputerScheduledScript(computerId, entityId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/ScheduledScripts/${entityId}`,
            method: 'get',
            params,
        });
    }
    getScheduledTaskList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/ScheduledTasks`,
            method: 'get',
            params,
        });
    }
    postScheduledTask(computerId, ComputerScheduledTask) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/ScheduledTasks`,
            method: 'post',
            data: ComputerScheduledTask,
        });
    }
    getComputerScriptHistoryList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/ScriptHistory`,
            method: 'get',
            params,
        });
    }
    getComputerSensorListAsync(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Sensors`,
            method: 'get',
            params,
        });
    }
    getComputerServiceList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Services`,
            method: 'get',
            params,
        });
    }
    getComputerSoftwareList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Software`,
            method: 'get',
            params,
        });
    }
    uninstallSoftware(computerId, softwareId, SoftwareUninstallInfo) {
        return this.request({
            path: `/api/v1/computers/${computerId}/software/${softwareId}/uninstall`,
            method: 'post',
            data: SoftwareUninstallInfo,
        });
    }
    getComputerSystemSlotList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/SystemSlots`,
            method: 'get',
            params,
        });
    }
    getComputerThirdPartyPatchListAsync(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/ThirdPartyPatches`,
            method: 'get',
            params,
        });
    }
    getComputerUpsListAsync(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/Ups`,
            method: 'get',
            params,
        });
    }
    getComputerVideoCardList(computerId, params = {}) {
        return this.request({
            path: `/api/v1/Computers/${computerId}/VideoCards`,
            method: 'get',
            params,
        });
    }
    getComputerChassis(params = {}) {
        return this.request({
            path: `/api/v1/Computers/Chassis`,
            method: 'get',
            params,
        });
    }
    getComputerDriveList(params = {}) {
        return this.request({
            path: `/api/v1/Computers/Drives`,
            method: 'get',
            params,
        });
    }
    getComputerMaintenanceModeList(params = {}) {
        return this.request({
            path: `/api/v1/Computers/MaintenanceModes`,
            method: 'get',
            params,
        });
    }
    getComputerMemorySlotList(params = {}) {
        return this.request({
            path: `/api/v1/Computers/MemorySlots`,
            method: 'get',
            params,
        });
    }
    getComputersSoftwareList(params = {}) {
        return this.request({
            path: `/api/v1/Computers/Software`,
            method: 'get',
            params,
        });
    }
    getRetiredAssetList(params = {}) {
        return this.request({
            path: `/api/v1/RetiredAssets`,
            method: 'get',
            params,
        });
    }
    classifyService(serviceId, ServiceClassification) {
        return this.request({
            path: `/api/v1/services/${serviceId}/classify`,
            method: 'post',
            data: ServiceClassification,
        });
    }
    getVirusScannerDefinitionList(params = {}) {
        return this.request({
            path: `/api/v1/VirusScannerDefs`,
            method: 'get',
            params,
        });
    }
    postVirusScannerDefinition(VirusScannerDef) {
        return this.request({
            path: `/api/v1/VirusScannerDefs`,
            method: 'post',
            data: VirusScannerDef,
        });
    }
}
exports.ComputersAPI = ComputersAPI;
