// connectwise-rest@1.3.0/dist/Automate/ContactsAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContactsAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Automate_1 = __importDefault(require("../Automate"));
/**
 * @module ContactsAPI
 */
/**
 * Contacts module
 * @public
 */
class ContactsAPI extends Automate_1.default {
    constructor(props) {
        super(props);
    }
    getContactList(params = {}) {
        return this.request({
            path: `/api/v1/Contacts`,
            method: 'get',
            params,
        });
    }
    getSystemContactList(params = {}) {
        return this.request({
            path: `/api/v1/SystemContacts`,
            method: 'get',
            params,
        });
    }
    getContactListV2(params = {}) {
        return this.request({
            path: `/api/v2/Contacts`,
            method: 'get',
            params,
        });
    }
    getContact(contactId, params = {}) {
        return this.request({
            path: `/api/v2/Contacts/${contactId}`,
            method: 'get',
            params,
        });
    }
}
exports.ContactsAPI = ContactsAPI;
