// connectwise-rest@1.3.0/dist/Automate/DrivesAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrivesAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Automate_1 = __importDefault(require("../Automate"));
/**
 * @module DrivesAPI
 */
/**
 * Drives module
 * @public
 */
class DrivesAPI extends Automate_1.default {
    constructor(props) {
        super(props);
    }
    getDailyDriveStats(driveId, params = {}) {
        return this.request({
            path: `/api/v1/drives/${driveId}/drivestats/daily`,
            method: 'get',
            params,
        });
    }
    getWeeklyDriveStats(driveId, params = {}) {
        return this.request({
            path: `/api/v1/drives/${driveId}/drivestats/weekly`,
            method: 'get',
            params,
        });
    }
    getYearlyDriveStats(driveId, params = {}) {
        return this.request({
            path: `/api/v1/drives/${driveId}/drivestats/yearly`,
            method: 'get',
            params,
        });
    }
    getMonthlyDriveStats(driveId, params = {}) {
        return this.request({
            path: `/api/v1/drives/${driveId}/drivestats/monthly`,
            method: 'get',
            params,
        });
    }
    getDriveStatisticsList(params = {}) {
        return this.request({
            path: `/api/v1/Statistics/Drives`,
            method: 'get',
            params,
        });
    }
}
exports.DrivesAPI = DrivesAPI;
