// connectwise-rest@1.3.0/dist/Automate/MonitorsAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MonitorsAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Automate_1 = __importDefault(require("../Automate"));
/**
 * @module MonitorsAPI
 */
/**
 * Monitors module
 * @public
 */
class MonitorsAPI extends Automate_1.default {
    constructor(props) {
        super(props);
    }
    getInternalMonitorResultList(params = {}) {
        return this.request({
            path: `/api/v1/InternalMonitorResults`,
            method: 'get',
            params,
        });
    }
    getMonitorHistoryList(params = {}) {
        return this.request({
            path: `/api/v1/MonitorHistory`,
            method: 'get',
            params,
        });
    }
    getDailyCollectedData(monitorId, params = {}) {
        return this.request({
            path: `/api/v1/Monitors/${monitorId}/CollectedData/DailyAverages`,
            method: 'get',
            params,
        });
    }
    getMonthlyCollectedData(monitorId, params = {}) {
        return this.request({
            path: `/api/v1/Monitors/${monitorId}/CollectedData/MonthlyAverages`,
            method: 'get',
            params,
        });
    }
    getWeeklyCollectedData(monitorId, params = {}) {
        return this.request({
            path: `/api/v1/Monitors/${monitorId}/CollectedData/WeeklyAverages`,
            method: 'get',
            params,
        });
    }
    getYearlyCollectedData(monitorId, params = {}) {
        return this.request({
            path: `/api/v1/Monitors/${monitorId}/CollectedData/YearlyAverages`,
            method: 'get',
            params,
        });
    }
    getDataCollectionSettingsForMonitorAsync(monitorId, params = {}) {
        return this.request({
            path: `/api/v1/Monitors/${monitorId}/DataCollectionSettings`,
            method: 'get',
            params,
        });
    }
    getMonitorStatisticsList(params = {}) {
        return this.request({
            path: `/api/v1/MonitorStatistics`,
            method: 'get',
            params,
        });
    }
    getSensorCheckList(params = {}) {
        return this.request({
            path: `/api/v1/SensorChecks`,
            method: 'get',
            params,
        });
    }
}
exports.MonitorsAPI = MonitorsAPI;
