// connectwise-rest@1.3.0/dist/Automate/NetworkDevicesAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkDevicesAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Automate_1 = __importDefault(require("../Automate"));
/**
 * @module NetworkDevicesAPI
 */
/**
 * NetworkDevices module
 * @public
 */
class NetworkDevicesAPI extends Automate_1.default {
    constructor(props) {
        super(props);
    }
    getNetworkDeviceList(params = {}) {
        return this.request({
            path: `/api/v1/NetworkDevices`,
            method: 'get',
            params,
        });
    }
    postNetworkDevice(NetworkDevice) {
        return this.request({
            path: `/api/v1/NetworkDevices`,
            method: 'post',
            data: NetworkDevice,
        });
    }
    getRouters(params = {}) {
        return this.request({
            path: `/api/v1/routers`,
            method: 'get',
            params,
        });
    }
}
exports.NetworkDevicesAPI = NetworkDevicesAPI;
