// connectwise-rest@1.3.0/dist/Automate/PatchingAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PatchingAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Automate_1 = __importDefault(require("../Automate"));
/**
 * @module PatchingAPI
 */
/**
 * Patching module
 * @public
 */
class PatchingAPI extends Automate_1.default {
    constructor(props) {
        super(props);
    }
    getApprovalPolicyListAsync(params = {}) {
        return this.request({
            path: `/api/v1/ApprovalPolicies`,
            method: 'get',
            params,
        });
    }
    getGroupPatchingPolicyListAsync(params = {}) {
        return this.request({
            path: `/api/v1/GroupPatchingPolicies`,
            method: 'get',
            params,
        });
    }
    getMicrosoftUpdatePolicyListAsync(params = {}) {
        return this.request({
            path: `/api/v1/MicrosoftUpdatePolicies`,
            method: 'get',
            params,
        });
    }
    deployAllApproved(PatchActionArgs) {
        return this.request({
            path: `/api/v1/PatchActions/DeployAllApproved`,
            method: 'post',
            data: PatchActionArgs,
        });
    }
    deployAllSecurity(PatchActionArgs) {
        return this.request({
            path: `/api/v1/PatchActions/DeployAllSecurity`,
            method: 'post',
            data: PatchActionArgs,
        });
    }
    reattemptFailed(PatchActionArgs) {
        return this.request({
            path: `/api/v1/PatchActions/ReattemptFailed`,
            method: 'post',
            data: PatchActionArgs,
        });
    }
    setToPilotStage(PatchActionArgs) {
        return this.request({
            path: `/api/v1/PatchActions/SetToPilotStage`,
            method: 'post',
            data: PatchActionArgs,
        });
    }
    setToProductionStage(PatchActionArgs) {
        return this.request({
            path: `/api/v1/PatchActions/SetToProductionStage`,
            method: 'post',
            data: PatchActionArgs,
        });
    }
    setToTestStage(PatchActionArgs) {
        return this.request({
            path: `/api/v1/PatchActions/SetToTestStage`,
            method: 'post',
            data: PatchActionArgs,
        });
    }
    getPatchHistoryListAsync(params = {}) {
        return this.request({
            path: `/api/v1/PatchHistory`,
            method: 'get',
            params,
        });
    }
    getThirdPartyUpdatePolicyListAsync(params = {}) {
        return this.request({
            path: `/api/v1/ThirdPartyUpdatePolicies`,
            method: 'get',
            params,
        });
    }
}
exports.PatchingAPI = PatchingAPI;
