// connectwise-rest@1.3.0/dist/Automate/RemoteAgentAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemoteAgentAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Automate_1 = __importDefault(require("../Automate"));
/**
 * @module RemoteAgentAPI
 */
/**
 * RemoteAgent module
 * @public
 */
class RemoteAgentAPI extends Automate_1.default {
    constructor(props) {
        super(props);
    }
    getRemoteAgentScheduleList(params = {}) {
        return this.request({
            path: `/api/v1/RemoteAgentSchedules`,
            method: 'get',
            params,
        });
    }
    postRemoteAgentSchedule(RemoteAgentSchedule) {
        return this.request({
            path: `/api/v1/RemoteAgentSchedules`,
            method: 'post',
            data: RemoteAgentSchedule,
        });
    }
    getRemoteAgentTemplateListAsync(params = {}) {
        return this.request({
            path: `/api/v1/RemoteAgentTemplates`,
            method: 'get',
            params,
        });
    }
    postRemoteAgentTemplateAsync(RemoteAgentTemplate) {
        return this.request({
            path: `/api/v1/RemoteAgentTemplates`,
            method: 'post',
            data: RemoteAgentTemplate,
        });
    }
}
exports.RemoteAgentAPI = RemoteAgentAPI;
