// connectwise-rest@1.3.0/dist/Automate/SearchesAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchesAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Automate_1 = __importDefault(require("../Automate"));
/**
 * @module SearchesAPI
 */
/**
 * Searches module
 * @public
 */
class SearchesAPI extends Automate_1.default {
    constructor(props) {
        super(props);
    }
    getSearchList(params = {}) {
        return this.request({
            path: `/api/v1/Searches`,
            method: 'get',
            params,
        });
    }
    getSearchFolderList(params = {}) {
        return this.request({
            path: `/api/v1/SearchFolders`,
            method: 'get',
            params,
        });
    }
    postSearchFolder(SearchFolder) {
        return this.request({
            path: `/api/v1/SearchFolders`,
            method: 'post',
            data: SearchFolder,
        });
    }
}
exports.SearchesAPI = SearchesAPI;
