// nestjs-cls@5.4.3/dist/src/lib/cls-module/cls-common.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClsCommonModule = void 0;
const common_1 = require("@nestjs/common");
const cls_service_manager_1 = require("../cls-service-manager");
const cls_service_1 = require("../cls.service");
const proxy_provider_1 = require("../proxy-provider");
const proxy_provider_manager_1 = require("../proxy-provider/proxy-provider-manager");
const clsServiceProvider = {
    provide: cls_service_1.ClsService,
    useValue: cls_service_manager_1.ClsServiceManager.getClsService(),
};
const commonProviders = [
    clsServiceProvider,
    ...[...proxy_provider_1.defaultProxyProviderTokens].map((token) => proxy_provider_manager_1.ProxyProviderManager.createProxyProviderFromExistingKey(token, {
        strict: true,
    })),
];
/**
 * ClsCommonModule provides common providers for the ClsModule and ClsRootModule.
 */
let ClsCommonModule = class ClsCommonModule {
};
exports.ClsCommonModule = ClsCommonModule;
exports.ClsCommonModule = ClsCommonModule = __decorate([
    (0, common_1.Module)({
        providers: [...commonProviders],
        exports: [...commonProviders],
    })
], ClsCommonModule);
