// connectwise-rest@1.3.0/dist/AutomateAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Automate_1 = __importDefault(require("./Automate"));
const AVTemplatePoliciesAPI_1 = require("./Automate/AVTemplatePoliciesAPI");
const ClientsAPI_1 = require("./Automate/ClientsAPI");
const CommandsAPI_1 = require("./Automate/CommandsAPI");
const ComputersAPI_1 = require("./Automate/ComputersAPI");
const ContactsAPI_1 = require("./Automate/ContactsAPI");
const DataViewsAPI_1 = require("./Automate/DataViewsAPI");
const DrivesAPI_1 = require("./Automate/DrivesAPI");
const EventLogsAPI_1 = require("./Automate/EventLogsAPI");
const LocationsAPI_1 = require("./Automate/LocationsAPI");
const MaintenanceWindowDefinitionsAPI_1 = require("./Automate/MaintenanceWindowDefinitionsAPI");
const MonitorsAPI_1 = require("./Automate/MonitorsAPI");
const NetworkDevicesAPI_1 = require("./Automate/NetworkDevicesAPI");
const PatchingAPI_1 = require("./Automate/PatchingAPI");
const RemoteAgentAPI_1 = require("./Automate/RemoteAgentAPI");
const ScriptingAPI_1 = require("./Automate/ScriptingAPI");
const SearchesAPI_1 = require("./Automate/SearchesAPI");
const SystemAPI_1 = require("./Automate/SystemAPI");
const TicketsAPI_1 = require("./Automate/TicketsAPI");
const UserProfilesAPI_1 = require("./Automate/UserProfilesAPI");
/**
 * @module AutomateAPI
 * @public
 */
class AutomateAPI extends Automate_1.default {
    constructor(options) {
        super(options);
        /**
         * @public
         */
        Object.defineProperty(this, "AVTemplatePoliciesAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ClientsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "CommandsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ComputersAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ContactsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "DataViewsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "DrivesAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "EventLogsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "LocationsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "MaintenanceWindowDefinitionsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "MonitorsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "NetworkDevicesAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "PatchingAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "RemoteAgentAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ScriptingAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "SearchesAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "SystemAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "TicketsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "UserProfilesAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this.AVTemplatePoliciesAPI = new AVTemplatePoliciesAPI_1.AVTemplatePoliciesAPI(options);
        this.ClientsAPI = new ClientsAPI_1.ClientsAPI(options);
        this.CommandsAPI = new CommandsAPI_1.CommandsAPI(options);
        this.ComputersAPI = new ComputersAPI_1.ComputersAPI(options);
        this.ContactsAPI = new ContactsAPI_1.ContactsAPI(options);
        this.DataViewsAPI = new DataViewsAPI_1.DataViewsAPI(options);
        this.DrivesAPI = new DrivesAPI_1.DrivesAPI(options);
        this.EventLogsAPI = new EventLogsAPI_1.EventLogsAPI(options);
        this.LocationsAPI = new LocationsAPI_1.LocationsAPI(options);
        this.MaintenanceWindowDefinitionsAPI = new MaintenanceWindowDefinitionsAPI_1.MaintenanceWindowDefinitionsAPI(options);
        this.MonitorsAPI = new MonitorsAPI_1.MonitorsAPI(options);
        this.NetworkDevicesAPI = new NetworkDevicesAPI_1.NetworkDevicesAPI(options);
        this.PatchingAPI = new PatchingAPI_1.PatchingAPI(options);
        this.RemoteAgentAPI = new RemoteAgentAPI_1.RemoteAgentAPI(options);
        this.ScriptingAPI = new ScriptingAPI_1.ScriptingAPI(options);
        this.SearchesAPI = new SearchesAPI_1.SearchesAPI(options);
        this.SystemAPI = new SystemAPI_1.SystemAPI(options);
        this.TicketsAPI = new TicketsAPI_1.TicketsAPI(options);
        this.UserProfilesAPI = new UserProfilesAPI_1.UserProfilesAPI(options);
    }
}
/**
 * @public
 */
exports.default = AutomateAPI;
