// connectwise-rest@1.3.0/dist/ManageAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Manage_1 = __importDefault(require("./Manage"));
const CompanyAPI_1 = require("./Manage/CompanyAPI");
const ConfigurationsAPI_1 = require("./Manage/ConfigurationsAPI");
const ExpenseAPI_1 = require("./Manage/ExpenseAPI");
const FinanceAPI_1 = require("./Manage/FinanceAPI");
const MarketingAPI_1 = require("./Manage/MarketingAPI");
const ProcurementAPI_1 = require("./Manage/ProcurementAPI");
const ProjectAPI_1 = require("./Manage/ProjectAPI");
const SalesAPI_1 = require("./Manage/SalesAPI");
const ScheduleAPI_1 = require("./Manage/ScheduleAPI");
const ServiceAPI_1 = require("./Manage/ServiceAPI");
const SystemAPI_1 = require("./Manage/SystemAPI");
const TimeAPI_1 = require("./Manage/TimeAPI");
/**
 * @public
 */
class ManageAPI extends Manage_1.default {
    constructor(options) {
        super(options);
        Object.defineProperty(this, "CompanyAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ConfigurationsAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ExpenseAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "FinanceAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "MarketingAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ProcurementAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ProjectAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "SalesAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ScheduleAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "ServiceAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "SystemAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "TimeAPI", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this.CompanyAPI = new CompanyAPI_1.CompanyAPI(options);
        this.ConfigurationsAPI = new ConfigurationsAPI_1.ConfigurationsAPI(options);
        this.ExpenseAPI = new ExpenseAPI_1.ExpenseAPI(options);
        this.FinanceAPI = new FinanceAPI_1.FinanceAPI(options);
        this.MarketingAPI = new MarketingAPI_1.MarketingAPI(options);
        this.ProcurementAPI = new ProcurementAPI_1.ProcurementAPI(options);
        this.ProjectAPI = new ProjectAPI_1.ProjectAPI(options);
        this.SalesAPI = new SalesAPI_1.SalesAPI(options);
        this.ScheduleAPI = new ScheduleAPI_1.ScheduleAPI(options);
        this.ServiceAPI = new ServiceAPI_1.ServiceAPI(options);
        this.SystemAPI = new SystemAPI_1.SystemAPI(options);
        this.TimeAPI = new TimeAPI_1.TimeAPI(options);
    }
}
/**
 * @public
 */
exports.default = ManageAPI;
