// connectwise-rest@1.3.0/dist/Manage/ConfigurationsAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigurationsAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Manage_1 = __importDefault(require("../Manage"));
/**
 * @module ConfigurationsAPI
 */
/**
 * Configurations module
 * @public
 */
class ConfigurationsAPI extends Manage_1.default {
    constructor(props) {
        super(props);
    }
    getConfigurationsTypesByGrandparentIdQuestionsByParentIdValuesByIdInfo(grandparentId, parentId, id, params = {}) {
        return this.request({
            path: `/configurations/types/${grandparentId}/questions/${parentId}/values/${id}/info`,
            method: 'get',
            params,
        });
    }
    getConfigurationsTypesByGrandparentIdQuestionsByParentIdValuesInfo(grandparentId, parentId, params = {}) {
        return this.request({
            path: `/configurations/types/${grandparentId}/questions/${parentId}/values/info`,
            method: 'get',
            params,
        });
    }
    getConfigurationsTypesByGrandparentIdQuestionsByParentIdValuesInfoCount(grandparentId, parentId, params = {}) {
        return this.request({
            path: `/configurations/types/${grandparentId}/questions/${parentId}/values/info/count`,
            method: 'get',
            params,
        });
    }
    getConfigurationsTypesByParentIdQuestionsByIdInfo(id, parentId, params = {}) {
        return this.request({
            path: `/configurations/types/${parentId}/questions/${id}/info`,
            method: 'get',
            params,
        });
    }
    getConfigurationsTypesByParentIdQuestionsInfo(parentId, params = {}) {
        return this.request({
            path: `/configurations/types/${parentId}/questions/info`,
            method: 'get',
            params,
        });
    }
    getConfigurationsTypesByParentIdQuestionsInfoCount(parentId, params = {}) {
        return this.request({
            path: `/configurations/types/${parentId}/questions/info/count`,
            method: 'get',
            params,
        });
    }
    getConfigurationsTypesInfo(params = {}) {
        return this.request({
            path: `/configurations/types/info`,
            method: 'get',
            params,
        });
    }
    getConfigurationsTypesInfoCount(params = {}) {
        return this.request({
            path: `/configurations/types/info/count`,
            method: 'get',
            params,
        });
    }
}
exports.ConfigurationsAPI = ConfigurationsAPI;
