// connectwise-rest@1.3.0/dist/Manage/ScheduleAPI.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleAPI = void 0;
/* This file was auto-generated, do not manually edit. */
const Manage_1 = __importDefault(require("../Manage"));
/**
 * @module ScheduleAPI
 */
/**
 * Schedule module
 * @public
 */
class ScheduleAPI extends Manage_1.default {
    constructor(props) {
        super(props);
    }
    getScheduleCalendars(params = {}) {
        return this.request({
            path: `/schedule/calendars`,
            method: 'get',
            params,
        });
    }
    postScheduleCalendars(calendar) {
        return this.request({
            path: `/schedule/calendars`,
            method: 'post',
            data: calendar,
        });
    }
    getScheduleCalendarsById(id, params = {}) {
        return this.request({
            path: `/schedule/calendars/${id}`,
            method: 'get',
            params,
        });
    }
    patchScheduleCalendarsById(id, patchOperations) {
        return this.request({
            path: `/schedule/calendars/${id}`,
            method: 'patch',
            data: patchOperations,
        });
    }
    putScheduleCalendarsById(id, calendar) {
        return this.request({
            path: `/schedule/calendars/${id}`,
            method: 'put',
            data: calendar,
        });
    }
    deleteScheduleCalendarsById(id) {
        return this.request({
            path: `/schedule/calendars/${id}`,
            method: 'delete',
        });
    }
    postScheduleCalendarsByIdCopy(id) {
        return this.request({
            path: `/schedule/calendars/${id}/copy`,
            method: 'post',
        });
    }
    getScheduleCalendarsByIdInfo(id, params = {}) {
        return this.request({
            path: `/schedule/calendars/${id}/info`,
            method: 'get',
            params,
        });
    }
    getScheduleCalendarsByIdUsages(id, params = {}) {
        return this.request({
            path: `/schedule/calendars/${id}/usages`,
            method: 'get',
            params,
        });
    }
    getScheduleCalendarsByIdUsagesList(id, params = {}) {
        return this.request({
            path: `/schedule/calendars/${id}/usages/list`,
            method: 'get',
            params,
        });
    }
    getScheduleCalendarsCount(params = {}) {
        return this.request({
            path: `/schedule/calendars/count`,
            method: 'get',
            params,
        });
    }
    getScheduleCalendarsInfo(params = {}) {
        return this.request({
            path: `/schedule/calendars/info`,
            method: 'get',
            params,
        });
    }
    getScheduleCalendarsInfoCount(params = {}) {
        return this.request({
            path: `/schedule/calendars/info/count`,
            method: 'get',
            params,
        });
    }
    getScheduleColors(params = {}) {
        return this.request({
            path: `/schedule/colors`,
            method: 'get',
            params,
        });
    }
    getScheduleColorsById(id, params = {}) {
        return this.request({
            path: `/schedule/colors/${id}`,
            method: 'get',
            params,
        });
    }
    putScheduleColorsById(id, scheduleColor) {
        return this.request({
            path: `/schedule/colors/${id}`,
            method: 'put',
            data: scheduleColor,
        });
    }
    patchScheduleColorsById(id, patchOperations) {
        return this.request({
            path: `/schedule/colors/${id}`,
            method: 'patch',
            data: patchOperations,
        });
    }
    postScheduleColorsByIdClear(id) {
        return this.request({
            path: `/schedule/colors/${id}/clear`,
            method: 'post',
        });
    }
    getScheduleColorsCount(params = {}) {
        return this.request({
            path: `/schedule/colors/count`,
            method: 'get',
            params,
        });
    }
    postScheduleColorsReset() {
        return this.request({
            path: `/schedule/colors/reset`,
            method: 'post',
        });
    }
    getScheduleDetails(params = {}) {
        return this.request({
            path: `/schedule/details`,
            method: 'get',
            params,
        });
    }
    getScheduleDetailsById(id, params = {}) {
        return this.request({
            path: `/schedule/details/${id}`,
            method: 'get',
            params,
        });
    }
    getScheduleDetailsCount(params = {}) {
        return this.request({
            path: `/schedule/details/count`,
            method: 'get',
            params,
        });
    }
    getScheduleEntries(params = {}) {
        return this.request({
            path: `/schedule/entries`,
            method: 'get',
            params,
        });
    }
    postScheduleEntries(scheduleEntry) {
        return this.request({
            path: `/schedule/entries`,
            method: 'post',
            data: scheduleEntry,
        });
    }
    getScheduleEntriesById(id, params = {}) {
        return this.request({
            path: `/schedule/entries/${id}`,
            method: 'get',
            params,
        });
    }
    deleteScheduleEntriesById(id) {
        return this.request({
            path: `/schedule/entries/${id}`,
            method: 'delete',
        });
    }
    putScheduleEntriesById(id, scheduleEntry) {
        return this.request({
            path: `/schedule/entries/${id}`,
            method: 'put',
            data: scheduleEntry,
        });
    }
    patchScheduleEntriesById(id, patchOperations) {
        return this.request({
            path: `/schedule/entries/${id}`,
            method: 'patch',
            data: patchOperations,
        });
    }
    getScheduleEntriesByParentIdDetails(parentId, params = {}) {
        return this.request({
            path: `/schedule/entries/${parentId}/details`,
            method: 'get',
            params,
        });
    }
    getScheduleEntriesByParentIdDetailsById(id, parentId, params = {}) {
        return this.request({
            path: `/schedule/entries/${parentId}/details/${id}`,
            method: 'get',
            params,
        });
    }
    getScheduleEntriesByParentIdDetailsCount(parentId, params = {}) {
        return this.request({
            path: `/schedule/entries/${parentId}/details/count`,
            method: 'get',
            params,
        });
    }
    getScheduleEntriesCount(params = {}) {
        return this.request({
            path: `/schedule/entries/count`,
            method: 'get',
            params,
        });
    }
    getScheduleHolidayLists(params = {}) {
        return this.request({
            path: `/schedule/holidayLists`,
            method: 'get',
            params,
        });
    }
    postScheduleHolidayLists(holidayList) {
        return this.request({
            path: `/schedule/holidayLists`,
            method: 'post',
            data: holidayList,
        });
    }
    getScheduleHolidayListsById(id, params = {}) {
        return this.request({
            path: `/schedule/holidayLists/${id}`,
            method: 'get',
            params,
        });
    }
    deleteScheduleHolidayListsById(id) {
        return this.request({
            path: `/schedule/holidayLists/${id}`,
            method: 'delete',
        });
    }
    putScheduleHolidayListsById(id, holidayList) {
        return this.request({
            path: `/schedule/holidayLists/${id}`,
            method: 'put',
            data: holidayList,
        });
    }
    patchScheduleHolidayListsById(id, patchOperations) {
        return this.request({
            path: `/schedule/holidayLists/${id}`,
            method: 'patch',
            data: patchOperations,
        });
    }
    getScheduleHolidaylistsByIdInfo(id, params = {}) {
        return this.request({
            path: `/schedule/holidaylists/${id}/info`,
            method: 'get',
            params,
        });
    }
    getScheduleHolidayListsByIdUsages(id, params = {}) {
        return this.request({
            path: `/schedule/holidayLists/${id}/usages`,
            method: 'get',
            params,
        });
    }
    getScheduleHolidayListsByIdUsagesList(id, params = {}) {
        return this.request({
            path: `/schedule/holidayLists/${id}/usages/list`,
            method: 'get',
            params,
        });
    }
    getScheduleHolidayListsByParentIdHolidays(parentId, params = {}) {
        return this.request({
            path: `/schedule/holidayLists/${parentId}/holidays`,
            method: 'get',
            params,
        });
    }
    postScheduleHolidayListsByParentIdHolidays(parentId, holiday) {
        return this.request({
            path: `/schedule/holidayLists/${parentId}/holidays`,
            method: 'post',
            data: holiday,
        });
    }
    getScheduleHolidayListsByParentIdHolidaysById(id, parentId, params = {}) {
        return this.request({
            path: `/schedule/holidayLists/${parentId}/holidays/${id}`,
            method: 'get',
            params,
        });
    }
    deleteScheduleHolidayListsByParentIdHolidaysById(id, parentId) {
        return this.request({
            path: `/schedule/holidayLists/${parentId}/holidays/${id}`,
            method: 'delete',
        });
    }
    putScheduleHolidayListsByParentIdHolidaysById(id, parentId, holiday) {
        return this.request({
            path: `/schedule/holidayLists/${parentId}/holidays/${id}`,
            method: 'put',
            data: holiday,
        });
    }
    patchScheduleHolidayListsByParentIdHolidaysById(id, parentId, patchOperations) {
        return this.request({
            path: `/schedule/holidayLists/${parentId}/holidays/${id}`,
            method: 'patch',
            data: patchOperations,
        });
    }
    getScheduleHolidayListsByParentIdHolidaysCount(parentId, params = {}) {
        return this.request({
            path: `/schedule/holidayLists/${parentId}/holidays/count`,
            method: 'get',
            params,
        });
    }
    postScheduleHolidayListsCopy(copy) {
        return this.request({
            path: `/schedule/holidayLists/copy`,
            method: 'post',
            data: copy,
        });
    }
    getScheduleHolidayListsCount(params = {}) {
        return this.request({
            path: `/schedule/holidayLists/count`,
            method: 'get',
            params,
        });
    }
    getScheduleHolidaylistsInfo(params = {}) {
        return this.request({
            path: `/schedule/holidaylists/info`,
            method: 'get',
            params,
        });
    }
    getSchedulePortalcalendars(params = {}) {
        return this.request({
            path: `/schedule/portalcalendars`,
            method: 'get',
            params,
        });
    }
    getSchedulePortalcalendarsById(id, params = {}) {
        return this.request({
            path: `/schedule/portalcalendars/${id}`,
            method: 'get',
            params,
        });
    }
    putSchedulePortalcalendarsById(id, portalCalendar) {
        return this.request({
            path: `/schedule/portalcalendars/${id}`,
            method: 'put',
            data: portalCalendar,
        });
    }
    patchSchedulePortalcalendarsById(id, patchOperations) {
        return this.request({
            path: `/schedule/portalcalendars/${id}`,
            method: 'patch',
            data: patchOperations,
        });
    }
    getSchedulePortalcalendarsCount(params = {}) {
        return this.request({
            path: `/schedule/portalcalendars/count`,
            method: 'get',
            params,
        });
    }
    getScheduleReminderTimes(params = {}) {
        return this.request({
            path: `/schedule/reminderTimes`,
            method: 'get',
            params,
        });
    }
    getScheduleReminderTimesById(id, params = {}) {
        return this.request({
            path: `/schedule/reminderTimes/${id}`,
            method: 'get',
            params,
        });
    }
    putScheduleReminderTimesById(id, reminderTime) {
        return this.request({
            path: `/schedule/reminderTimes/${id}`,
            method: 'put',
            data: reminderTime,
        });
    }
    patchScheduleReminderTimesById(id, patchOperations) {
        return this.request({
            path: `/schedule/reminderTimes/${id}`,
            method: 'patch',
            data: patchOperations,
        });
    }
    getScheduleReminderTimesCount(params = {}) {
        return this.request({
            path: `/schedule/reminderTimes/count`,
            method: 'get',
            params,
        });
    }
    getScheduleStatuses(params = {}) {
        return this.request({
            path: `/schedule/statuses`,
            method: 'get',
            params,
        });
    }
    postScheduleStatuses(scheduleStatus) {
        return this.request({
            path: `/schedule/statuses`,
            method: 'post',
            data: scheduleStatus,
        });
    }
    getScheduleStatusesById(id, params = {}) {
        return this.request({
            path: `/schedule/statuses/${id}`,
            method: 'get',
            params,
        });
    }
    deleteScheduleStatusesById(id) {
        return this.request({
            path: `/schedule/statuses/${id}`,
            method: 'delete',
        });
    }
    putScheduleStatusesById(id, scheduleStatus) {
        return this.request({
            path: `/schedule/statuses/${id}`,
            method: 'put',
            data: scheduleStatus,
        });
    }
    patchScheduleStatusesById(id, patchOperations) {
        return this.request({
            path: `/schedule/statuses/${id}`,
            method: 'patch',
            data: patchOperations,
        });
    }
    getScheduleStatusesCount(params = {}) {
        return this.request({
            path: `/schedule/statuses/count`,
            method: 'get',
            params,
        });
    }
    getScheduleTypes(params = {}) {
        return this.request({
            path: `/schedule/types`,
            method: 'get',
            params,
        });
    }
    postScheduleTypes(scheduleType) {
        return this.request({
            path: `/schedule/types`,
            method: 'post',
            data: scheduleType,
        });
    }
    getScheduleTypesById(id, params = {}) {
        return this.request({
            path: `/schedule/types/${id}`,
            method: 'get',
            params,
        });
    }
    deleteScheduleTypesById(id) {
        return this.request({
            path: `/schedule/types/${id}`,
            method: 'delete',
        });
    }
    putScheduleTypesById(id, scheduleType) {
        return this.request({
            path: `/schedule/types/${id}`,
            method: 'put',
            data: scheduleType,
        });
    }
    patchScheduleTypesById(id, patchOperations) {
        return this.request({
            path: `/schedule/types/${id}`,
            method: 'patch',
            data: patchOperations,
        });
    }
    getScheduleTypesByIdInfo(id, params = {}) {
        return this.request({
            path: `/schedule/types/${id}/info`,
            method: 'get',
            params,
        });
    }
    getScheduleTypesByIdUsages(id, params = {}) {
        return this.request({
            path: `/schedule/types/${id}/usages`,
            method: 'get',
            params,
        });
    }
    getScheduleTypesByIdUsagesList(id, params = {}) {
        return this.request({
            path: `/schedule/types/${id}/usages/list`,
            method: 'get',
            params,
        });
    }
    getScheduleTypesCount(params = {}) {
        return this.request({
            path: `/schedule/types/count`,
            method: 'get',
            params,
        });
    }
    getScheduleTypesInfo(params = {}) {
        return this.request({
            path: `/schedule/types/info`,
            method: 'get',
            params,
        });
    }
    getScheduleTypesInfoCount(params = {}) {
        return this.request({
            path: `/schedule/types/info/count`,
            method: 'get',
            params,
        });
    }
}
exports.ScheduleAPI = ScheduleAPI;
