// nestjs-cls@5.4.3/dist/src/lib/plugin/cls-plugin.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createClsPluginModule = createClsPluginModule;
const common_1 = require("@nestjs/common");
function createClsPluginModule(plugin) {
    var ClsPluginModule_1;
    let ClsPluginModule = ClsPluginModule_1 = class ClsPluginModule {
        static forRoot() {
            return {
                module: ClsPluginModule_1,
                imports: plugin.imports,
                providers: plugin.providers,
                exports: plugin.exports,
            };
        }
        async onModuleInit() {
            await plugin.onModuleInit?.();
        }
        async onModuleDestroy() {
            await plugin.onModuleDestroy?.();
        }
    };
    ClsPluginModule = ClsPluginModule_1 = __decorate([
        (0, common_1.Global)()
    ], ClsPluginModule);
    return ClsPluginModule.forRoot();
}
